core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/keystore/persistence/KeyStoreRegistryPersistenceManager.java

```diff
@@ -0,0 +1,421 @@
+    private static final String PROP_TENANT_PUB_KEY_FILE_NAME_APPENDER = "tenant.pub.key.file.name.appender";
+
+    /**
+     * This method encrypts the given password using the default crypto util.
+     *
+     * @param password Password to be encrypted.
+     * @return encrypted password.
+     */
+    public static String encryptPassword(String password) {
+
+        try {
+            return CryptoUtil.getDefaultCryptoUtil().encryptAndBase64Encode(password.getBytes());
+        } catch (CryptoException e) {
+            throw new SecurityException("Error encrypting the password", e);
+        }
+    }
+
+    /**
+     * This method decrypts the given encrypted password using the default crypto util.
+     *
+     * @param encryptedPassword Encrypted password.
+     * @return decrypted password as a character array..
+     */
+    public static char[] decryptPassword(String encryptedPassword) {
+
+        try {
+            CryptoUtil cryptoUtil = CryptoUtil.getDefaultCryptoUtil();
+            return new String(cryptoUtil.base64DecodeAndDecrypt(encryptedPassword)).toCharArray();
+        } catch (CryptoException e) {
+            throw new SecurityException("Error decrypting the password", e);
+        }
+    }
+
+    /**
+     * Add tenant's public certificate to the database.
+     *
+     * @param keyStoreName Name of the key store.
+     * @param publicCert   Public certificate of the tenant.
+     * @throws SecurityException If an error occurs while adding the tenant's public certificate.
+     */
+    private void addTenantPublicKey(String keyStoreName, X509Certificate publicCert, Registry registry)
+            throws SecurityException {
+
+        try {
+            // Create the public key resource.
+            Resource pubKeyResource = registry.newResource();
+            pubKeyResource.setContent(publicCert.getEncoded());
+            pubKeyResource.addProperty(PROP_TENANT_PUB_KEY_FILE_NAME_APPENDER, generatePublicCertId());
+            registry.put(RegistryResources.SecurityManagement.TENANT_PUBKEY_RESOURCE, pubKeyResource);
+
+            // Associate the public key with the keystore.
+            registry.addAssociation(RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName,
+                    RegistryResources.SecurityManagement.TENANT_PUBKEY_RESOURCE, ASSOCIATION_TENANT_KS_PUB_KEY);
+        } catch (RegistryException | CertificateEncodingException e) {
+            throw new SecurityException("Error when writing the keystore public cert to registry for keystore: " +
+                    keyStoreName, e);
+        }
+    }
+
+    /**
+     * Generates an ID for the public certificate, which is used as a file name suffix for the certificate.
+     * e.g. If keystore name is 'example-com.jks', public cert name will be 'example-com-343743.cert'.
+     *
+     * @return generated id to be used as a file name appender.
+     */
+    private String generatePublicCertId() {
```

Comment:
We can make this method static

Reply:
Any reasons for make this static?

Reply:
I thinks it's better to make a method static if we are can. @darshanasbg WDYT?

Reply:
> Any reasons for make this static?

If these are falls into the utils category, we can/better make them private static.. So +1 for the @hwupathum's suggestion..

Reply:
But this is a private method right?

Reply:
> But this is a private method right?

Yes. 😄

Reply:
its a private util. So we can make it private static

Reply:
Addressed with https://github.com/wso2/carbon-kernel/compare/6c4a7e5fc5efca69a114e5d3a08289b8dc809be7..3a9a8a70b486cbcffcd4d4da8c10588bd93f0358